gen/field.go

```diff
@@ -30,6 +30,7 @@ var reservedIdentifiers = map[string]struct{}{
 	"ToWire":   {},
 	"FromWire": {},
 	"String":   {},
+	"Equals":   {},
```

Comment:
I don't think we have existing tests for this. Just create a struct somewhere
with a `equals` field in one of the testdata Thrift files.